content/sample/images.md

```diff
@@ -1,26 +1,173 @@
 ---
-title: Responsive Images
-sub: Using nunjucks
+title: Image Layout Helpers
+sub: Using nunjucks and the image utility classes
 ---
 
 {% import 'embed.macros.njk' as embed %}
 
-## srcSet
 
-### embed (no width):
+{{ embed.figure(
+  data=[{
+    img: 'blog/2023/graph.jpg',
+    alt: 'sketched graph on paper'
+  }],
+  class='media-pull',
+  attrs={'style': '--rowspan: span 2'}
+) }}
+
+### `media-pull` will position an image next to the text on wide enough screens
+
+The `media-pull` class can be used to pull an image to the left of the content
+at the `xlarge` screen size. Using a style attribute, you can define how many
+rows to span using the css custom property `--rowspan`. The first image example
+uses the `embed.figure` macro.
+
 {{ embed.img(
-  src='blog/2016/trick-loops.jpg'
+  src='blog/2023/graph.jpg',
+  alt='A collection of button screenshots from one project',
+  attrs={
+    'class': 'media-pull',
+    'style': '--rowspan: span 2'
+  }
 ) }}
 
-### embed (width: 4410):
+### `media-pull` with the `embed.img` macro
+
+When using the `embed.img` macro in a markdown file, the image is placed
+within a `<p>` tag. We have a CSS helper that targets this nested image with
+`display: contents` in order to add layout classes which will work in [most
+modern browsers](https://caniuse.com/css-has). When unsupported, the image will
+appear centered within the content column.
+
+
+## Aligning images within the content areas
+
+If you want an image to float to a side of the text, you need to first wrap the
+image and text in a div with a class of `contain`.
+
+<div class="contain">
 {{ embed.img(
-  src='talks/mia-btconf.jpg',
-  alt='Miriam presenting'
+  src='work/coachhub/browse-coach-sitemap.jpg',
+  alt='site map for browsing coaches',
+  attrs={
+    'class': 'align-left'
+  }
+) }}
+
+To align an image to the left of the text, add the class `align-left`.
+Images will only float if the screen is wide enough.
+</div>
+
+<div class="contain">
+{{ embed.img(
+  src='work/coachhub/browse-coach-sitemap.jpg',
+  alt='site map for browsing coaches',
+  attrs={
+    'class': 'align-right'
+  }
 ) }}
 
-### sizes
+To align an image to the right of the text, add the class `align-right`.
+Images will only float if the screen is wide enough.
+</div>
+
+
+<div class="contain">
 {{ embed.img(
-  src='talks/mia-smashing19-rad.jpg',
+  src='work/coachhub/browse-coach-sitemap.jpg',
+  alt='site map for browsing coaches',
+  attrs={
+    'class': 'align-center'
+  }
+) }}
+
+To center align an image add the class `align-center`. This is only applicable
+if your image is smaller than the column width.
+
+</div>
+
+## Extending images outside the content column
+
+To extend an image further outside of the content column's width, add one of
+the following classes: `extend-small`, `extend-large`, or `extend-full`.
```

Comment:
@mirisuzanne I added `extend-small` back in this documentation since we can add this to the embed.img macro now. Also it is useful for showing how extend-small's width is the same as the embed.figure macro by default so it only makes sense to use it on embed.img.